### PR TITLE
Issue #793 - Hide root & replication users password from Postgres logs

### PR DESCRIPTION
**Pull request by keierk01**
* This change disable Postgres to print out password to logs even log_statement is something else than none (default)

Below is explained what happens when user is created or altered
1. Transaction is created when starting to manipulate password for the user
2. Logs are disabled by setting `log_statement` to `none` during the transaction

Because logs are disable during the transaction, passwords are not visible in logs at all.

### CODE REVIEW

**Comment by keierk01** [comment]
Could somebody restart the failing tests. I can't do that by myself. Thanks in advance.

**Comment by keierk01** [comment]
@sgotti It looks that there are some problems in your build system, because all recently pushed PR's are failing. Including this one. Can you help me to get this proceeding?

Thanks!

File: internal/postgresql/utils.go [hunk 1]
```diff
@@ -68,8 +68,23 @@ func setPassword(ctx context.Context, connParams ConnParams, username, password
...
-	_, err = dbExec(ctx, db, fmt.Sprintf(`alter role "%s" with password '%s';`, username, password))
-	return err
+	tx, err := db.Begin()
+	if err != nil {
+		return err
+	}
+
+	query := fmt.Sprintf("set local log_statement = %s", pq.QuoteLiteral("none"))
+	if _, err = tx.ExecContext(ctx, query); err != nil {
+		err = tx.Rollback()
```
Comment (sgotti) [inline internal/postgresql/utils.go:78]: assigning tx.Rollback to error will override the real error. I'll just ignore the Rollback error.

**Review by sgotti — CHANGES_REQUESTED** [review]
@keierk01 Sorry but I lost your PR. Some comments inline.

The build system is working. There're just some flakes due to big resources required by the tests that are running in parallel to speed them up. Nothing to worry about.

File: internal/postgresql/utils.go (hunk 1 above)
Comment (keierk01) [reply internal/postgresql/utils.go]: @sgotti That's true. I'm not so familiar with Go, but I agree with you. I updated the PR already.

**Comment by sgotti** [comment]
@keierk01 Thanks a lot! Merging.

**Comment by luomengY** [comment]
When will v0.18.0 be released?

**Pull request merged**